packages/internal/test/test-utils.js

```diff
@@ -71,10 +71,14 @@ test('assertAllDefined', t => {
   /** @type {{s: string, m?: string | undefined}} */
```

Comment:
I think you should remove the `?` but keep the `| undefined`, in which case, the `obj.m.length` access below ought to be legal

```suggestion
  /** @type {{s: string, m: string | undefined}} */
```

Reply:
Not any longer, with the removal of the `@returns` line. It would with your typedef,
```
/**
 * @template {Record<string, unknown>} T
 * @typedef {{[P in keyof T]: Exclude<T[P], undefined>;}} AllDefined
 */
```
If you're requesting that, I'll do it. But I'd just as soon leave out any type assertions.

Reply:
Ah I see. It's a little strange to have an `assertsFoo` function that doesn't have the equivalent type assertion, but if you're happy with the type of the asserted thing not being narrowed, then it works for me too.